File: Notification-Service/src/main/java/byrd/riley/notificationservice/model/Emailer.java
package byrd.riley.notificationservice.model;

import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;

public class Emailer {

	private final JavaMailSender emailSender;
	
	public Emailer(JavaMailSender emailSender) {
		this.emailSender = emailSender;
	}
	
	public void sendEmail(String receiver, String subject, String messageBody) {
		SimpleMailMessage message = new SimpleMailMessage();
		message.setFrom("noreply@techbill.org");
		message.setTo(receiver);
		message.setSubject(subject);
		message.setText(messageBody);
		emailSender.send(message);
	}
}

File: Notification-Service-Kotlin/src/main/kotlin/byrd/riley/notificationservicekotlin/model/Emailer.kt
package byrd.riley.notificationservicekotlin.model

import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender

class Emailer(private val emailSender: JavaMailSender) {
    fun sendEmail(receiver: String, subject: String, messageBody: String) {
        val message = SimpleMailMessage().apply {
            setFrom("noreply@example.com")
            setTo(receiver)
            setSubject(subject)
            setText(messageBody)
        }

        emailSender.send(message)
    }
}